IHP/SchemaCompiler.hs

```diff
@@ -444,38 +454,36 @@ toDefaultValueExpr Column { columnType, notNull, defaultValue = Just theDefaultV
                             _ -> "def"
 toDefaultValueExpr _ = "def"
 
-compileHasTableNameInstance :: (?schema :: Schema) => Statement -> Text
+compileHasTableNameInstance :: (?schema :: Schema) => CreateTable -> Text
 compileHasTableNameInstance table@(CreateTable { name }) =
     "type instance GetTableName (" <> tableNameToModelName name <> "' " <> unwords (map (const "_") (dataTypeArguments table)) <>  ") = " <> tshow name <> "\n"
     <> "type instance GetModelByTableName " <> tshow name <> " = " <> tableNameToModelName name <> "\n"
 
-compilePrimaryKeyInstance :: (?schema :: Schema) => Statement -> Text
-compilePrimaryKeyInstance table@(CreateTable { name, columns }) = "type instance PrimaryKey " <> tshow name <> " = " <> idType <> "\n"
+compilePrimaryKeyInstance :: (?schema :: Schema) => CreateTable -> Text
+compilePrimaryKeyInstance table@(CreateTable { name, columns, constraints }) =
+    "type instance PrimaryKey " <> tshow name <> " = " <> idType <> "\n"
     where
-        idColumn :: Column
-        (Just idColumn) = find (get #primaryKey) columns
-
         idType :: Text
-        idType = case get #columnType idColumn of
-                PUUID -> "UUID"
-                PSerial -> "Int"
-                PBigserial -> "Integer"
-                otherwise -> error ("Unexpected type for primary key column in table" <> cs name)
+        idType = case primaryKeyColumns table of
+            [] -> error "Impossible happened in compilePrimaryKeyInstance"
```

Comment:
Maybe we can improve this error message a little bit `Impossible happened in compilePrimaryKeyInstance. No primary keys found for table $tableName. Atleast one primary key is required.`. The impossible always happens at some point :D 

Btw: I've written down a few notes on IHP error message design goals here https://github.com/digitallyinduced/ihp/wiki/Design-Goals an hour ago. Let's strive to be the haskell package with the best error messages 👍

Reply:
@mpscholten Hmm, I've actually found out that SQL doesn't require primary keys on all tables. Is that something we want to allow?

Reply:
For now it's easier to just disallow this. Maybe we can skip tables without primary key in the code generation. This way you can still define tables without PK in your `Schema.sql`, but have to use plain sql queries to access them (like using  `sqlQuery "SELECT * FROM my_table_without_pk"`)